crypto/src/signature.rs

```diff
@@ -198,52 +175,53 @@ impl<T> SignatureOf<T> {
         }
     }
 
-    /// Creates new [`SignatureOf`] by signing value via [`KeyPair::private_key`].
+    /// Verifies signature for this hash
+    ///
     /// # Errors
-    /// Fails if decoding digest of key pair fails
-    pub fn from_hash(key_pair: KeyPair, hash: &HashOf<T>) -> Result<Self> {
-        Ok(Self(Signature::new(key_pair, hash.as_ref())?, PhantomData))
+    /// Fails if verification fails
+    #[cfg(feature = "std")]
+    pub fn verify_hash(&self, hash: &HashOf<T>) -> Result<(), Error> {
+        self.0.verify(hash.as_ref())
     }
 }
 
+#[cfg(feature = "std")]
 impl<T: Encode> SignatureOf<T> {
-    /// Creates new [`SignatureOf`] by signing value via [`KeyPair::private_key`].
+    /// Creates new [`SignatureOf`] by signing value via [`KeyPair::private_key`]
+    ///
     /// # Errors
     /// Fails if decoding digest of key pair fails
-    pub fn new(key_pair: KeyPair, value: &T) -> Result<Self> {
+    pub fn new(key_pair: KeyPair, value: &T) -> Result<Self, Error> {
         Self::from_hash(key_pair, &HashOf::new(value))
     }
 
     /// Verifies signature for this item
+    ///
     /// # Errors
     /// Fails if verification fails
-    pub fn verify(&self, value: &T) -> Result<(), SignatureVerificationFail<T>> {
+    pub fn verify(&self, value: &T) -> Result<(), Error> {
         self.verify_hash(&HashOf::new(value))
     }
 }
 
-impl<T> Clone for SignatureOf<T> {
-    fn clone(&self) -> Self {
-        Self(self.0.clone(), PhantomData)
-    }
-}
-
-impl<T> IntoSchema for SignatureOf<T> {
-    fn type_name() -> String {
-        Signature::type_name()
-    }
```

Comment:
Removed this method because it's inconsistent with `HashOf`.
Previously this serialized to:
```
iroha_schema::iroha_crypto::signature::SignatureOf
```

Now it serializes to (for every `T`):
```
iroha_schema::iroha_crypto::signature::SignatureOf<T>
```